lib/msf/base/sessions/ssh_command_shell_bind.rb

```diff
@@ -238,6 +238,11 @@ def initialize(ssh_connection, opts = {})
     def bootstrap(datastore = {}, handler = nil)
       # this won't work after the rstream is initialized, so do it first
       @platform = Metasploit::Framework::Ssh::Platform.get_platform(ssh_connection)
+      if @platform == 'windows'
+        extend(Msf::Sessions::WindowsEscaping)
+      else
+        extend(Msf::Sessions::UnixEscaping)
+      end
```

Comment:
Are you sure there are no other possibilities besides Windows and Unix wrt. shell escaping?

Reply:
From https://github.com/rapid7/metasploit-framework/blob/master/lib/metasploit/framework/ssh/platform.rb#L97-L132, I'm really not all that familiar with what exactly to expect shell-escaping-wise when you get (for example) an "arista" shell. If it's some custom shell completely unrelated to anything POSIX, then any of the behaviour that uses `escape_arg` (such as uploading files or launching processes) would be useless anyway and would need its entirely new implementation.

In those cases, I'm not sure what the best default would be. The alternative I guess would be better in those unknown cases to just not extend the object, and do no escaping.

Reply:
From my very brief analysis, the current state should easily cover all platforms except **Cisco ios**, **Juniper**, **Mikrotik**, **Arista** and **vcenter** - I'm not 100% sure how those shells work when it comes to character escaping. So the options are: add `else` for all above mentioned and just implement escaping for them later on with more thorough analysis or ignore it, hope for best and add at least `else` block for **Unknown** platform. We will probably have to create/extend tests for SSH sessions (though I'm not very familiar how they are implemented now, but I don't expect such cases are covered there).